FV3/atmos_cubed_sphere/tools/coarse_grained_restart_files.F90

```diff
@@ -0,0 +1,366 @@
+module coarse_grained_restart_files_mod
+
+  use coarse_graining_mod, only: compute_mass_weights, get_coarse_array_bounds,&
+       get_fine_array_bounds, MODEL_LEVEL, weighted_block_average, &
+       weighted_block_edge_average_x, weighted_block_edge_average_y
+  use field_manager_mod, only: MODEL_ATMOS
+  use fms_io_mod,      only: register_restart_field, save_restart
+  use fv_arrays_mod, only: coarse_restart_type, fv_atmos_type, fv_coarse_grid_bounds_type, fv_grid_bounds_type
+  use mpp_domains_mod, only: domain2d, EAST, NORTH
+  use tracer_manager_mod, only: get_tracer_names, set_tracer_profile
+
+  implicit none
+  private
+
+  public :: deallocate_coarse_restart_type, fv_coarse_restart_init, fv_io_write_restart_coarse
+
+  ! Global variables for this module, initialized in fv_coarse_restart_init
+  integer :: is, ie, js, je, npz
+  integer :: is_coarse, ie_coarse, js_coarse, je_coarse
+  integer :: n_prognostic_tracers, n_diagnostic_tracers, n_tracers
+
+contains
+
+  subroutine fv_coarse_restart_init(tile_count, nz, nt_prog, &
+       nt_phys, hydrostatic, hybrid_z, agrid_vel_rst, fv_land, &
+       coarse_domain, fine_bd, coarse_bd, restart)
+    integer, intent(in) :: tile_count, nz, nt_prog, nt_phys
+    logical, intent(in) :: hydrostatic, hybrid_z, agrid_vel_rst, fv_land
+    type(domain2d), intent(inout) :: coarse_domain
+    type(fv_grid_bounds_type), intent(in) :: fine_bd
+    type(fv_coarse_grid_bounds_type), intent(in) :: coarse_bd
+    type(coarse_restart_type), intent(inout) :: restart
+
+    call get_fine_array_bounds(fine_bd, is, ie, js, je)
+    call get_coarse_array_bounds(coarse_bd, is_coarse, ie_coarse, js_coarse, je_coarse)
+    n_prognostic_tracers = nt_prog
+    n_diagnostic_tracers = nt_phys
+    n_tracers = nt_prog + nt_phys
+    npz = nz
+
+    call allocate_coarse_restart_type(hydrostatic, hybrid_z, &
+         agrid_vel_rst, fv_land, restart)
+    call register_coarse_restart_files(tile_count, hydrostatic, &
+         hybrid_z, agrid_vel_rst, fv_land, &
+         coarse_domain, restart)
+  end subroutine fv_coarse_restart_init
+
+  subroutine fv_io_write_restart_coarse(Atm, grids_on_this_pe, timestamp)
+    type(fv_atmos_type), intent(inout) :: Atm(:)
+    logical, intent(in) :: grids_on_this_pe(:)
+    character(len=*), optional, intent(in) :: timestamp
+
+    integer :: tile_count, n_tiles
+
+    n_tiles = size(Atm(:))
+
+    do tile_count = 1, n_tiles
+       if (.not. grids_on_this_pe(tile_count)) cycle
```

Comment:
`cycle` skips the rest of this iteration of the `do` loop?

Reply:
Exactly.  This is mostly copied from the code that writes the full resolution restart files:

https://github.com/VulcanClimateModeling/fv3gfs-fortran/blob/42f2877a4e04882a1046d35662bce61c85385e27/FV3/atmos_cubed_sphere/tools/fv_io.F90#L595-L614